Manifest.toml

```diff
@@ -166,10 +172,10 @@ uuid = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 uuid = "56ddb016-857b-54e1-b83d-db4d58db5568"
 
 [[MacroTools]]
-deps = ["Compat", "DataStructures", "Test"]
-git-tree-sha1 = "82921f0e3bde6aebb8e524efc20f4042373c0c06"
+deps = ["CSTParser", "Compat", "DataStructures", "Test", "Tokenize"]
+git-tree-sha1 = "d6e9dedb8c92c3465575442da456aec15a89ff76"
 uuid = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
-version = "0.5.2"
```

Comment:
Wonder what caused this downgrade

Reply:
Shouldn't `up` upgrade MacroTools?

```
(NewPkgEval) pkg> st
Project NewPkgEval v0.1.0
    Status `~/Julia/pkg/NewPkgEval/Project.toml`
  [12aac903] BinaryBuilder v0.2.0 #master (https://github.com/JuliaPackaging/BinaryBuilder.jl.git)
  [ade2ca70] Dates 
  [76f85450] LibGit2 
  [44cfe95a] Pkg 
  [ea8e919c] SHA 

(NewPkgEval) pkg> up
  Updating registry at `~/Julia/depot/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Updating git-repo `https://github.com/JuliaPackaging/BinaryBuilder.jl.git`
 Resolving package versions...
  Updating `~/Julia/pkg/NewPkgEval/Project.toml`
  [12aac903] ~ BinaryBuilder v0.2.0 #master (https://github.com/JuliaPackaging/BinaryBuilder.jl.git)
  Updating `~/Julia/pkg/NewPkgEval/Manifest.toml`
  [12aac903] ~ BinaryBuilder v0.2.0 #master (https://github.com/JuliaPackaging/BinaryBuilder.jl.git)
  [07c12ed4] + ghr_jll v0.13.0+0

(NewPkgEval) pkg> add MacroTools@0.5.2
 Resolving package versions...
  Updating `~/Julia/pkg/NewPkgEval/Project.toml`
  [1914dd2f] + MacroTools v0.5.2
  Updating `~/Julia/pkg/NewPkgEval/Manifest.toml`
  [00ebfdb7] - CSTParser v1.0.0
  [34da2185] ↓ Compat v3.0.0 ⇒ v2.2.0
  [1914dd2f] ↑ MacroTools v0.5.1 ⇒ v0.5.2
  [0796e94c] - Tokenize v0.5.6

(NewPkgEval) pkg> rm MacroTools
  Updating `~/Julia/pkg/NewPkgEval/Project.toml`
  [1914dd2f] - MacroTools v0.5.2
  Updating `~/Julia/pkg/NewPkgEval/Manifest.toml`
 [no changes]

(NewPkgEval) pkg> resolve
 Resolving package versions...
  Updating `~/Julia/pkg/NewPkgEval/Project.toml`
 [no changes]
  Updating `~/Julia/pkg/NewPkgEval/Manifest.toml`
 [no changes]

(NewPkgEval) pkg> st -m
Project NewPkgEval v0.1.0
    Status `~/Julia/pkg/NewPkgEval/Manifest.toml`
  [1914dd2f] MacroTools v0.5.2
```

Reply:
Ah wait that downgraded Compat, so it's one or the other:

```
ERROR: Unsatisfiable requirements detected for package Compat [34da2185]:
 Compat [34da2185] log:
 ├─possible versions are: [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0, 3.0.0] or uninstalled
 ├─restricted to versions 3.0.0 by an explicit requirement, leaving only versions 3.0.0
 └─restricted by compatibility requirements with MacroTools [1914dd2f] to versions: [2.0.0, 2.1.0, 2.2.0] — no versions left
   └─MacroTools [1914dd2f] log:
     ├─possible versions are: [0.4.3-0.4.5, 0.5.0-0.5.2] or uninstalled
     └─restricted to versions 0.5.2 by an explicit requirement, leaving only versions 0.5.2
```

Reply:
```
  [00ebfdb7] - CSTParser v1.0.0
  [34da2185] ↓ Compat v3.0.0 ⇒ v2.2.0
  [1914dd2f] ↑ MacroTools v0.5.1 ⇒ v0.5.2
  [0796e94c] - Tokenize v0.5.6
```

So we have to chose between Compat and MacroTools and the resolver can do whatever it wants.

Reply:
https://github.com/MikeInnes/MacroTools.jl/pulls